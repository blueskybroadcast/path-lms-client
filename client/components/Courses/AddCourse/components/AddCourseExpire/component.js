import React from 'react';
import PropTypes from 'prop-types';
import DatePicker from 'react-datepicker';

const AddCourseExpire = ({
  expirable, repurchasable, expirationByDate, expirationDate, expirationByDays, expirationDays,
  handleChange, handleExpirationTypeChange, handleExpirationDateChange
}) => (
  <fieldset data-expirable="false" id="expirable_field">
    <legend>
      <span>Expiry Data</span>
    </legend>
    <div className="control-group featured">
      <label htmlFor="index_course_expiration_expirable">
        Expires
      </label>
      <div className="controls">
        <label
          htmlFor="index_course_expiration_expirable"
          className="label-switch"
          role="checkbox"
          tabIndex="0"
          aria-checked="true"
        >
          <input
            checked={expirable}
            onChange={handleChange}
            type="checkbox"
            name="expirable"
            id="index_course_expiration_expirable"
            data-role="expiration_expirable"
          />
          <div className="checkbox" />
        </label>
        <p className="help-block">Access to this item expires</p>
      </div>
    </div>
    { expirable && (
      <div id="expiration_strategies_fields">
        <div className="control-group featured">
          <label htmlFor="index_course_expiration_repurchasable">
            Repurchasable
          </label>
          <div className="controls">
            <label
              htmlFor="index_course_expiration_repurchasable"
              className="
              label-switch"
              role="checkbox"
              tabIndex="0"
              aria-checked="true"
            >
              <input
                checked={repurchasable}
                onChange={handleChange}
                type="checkbox"
                name="repurchasable"
                id="index_course_expiration_repurchasable"
              />
              <div className="checkbox" />
            </label>
            <p className="help-block">This item can be repurchased after expiration</p>
          </div>
        </div>
        <div className="control-group featured expiration-strategy">
          <div className="controls">
            <label
              htmlFor="index_course_expiration_expires_at_strategy"
              className="label-switch"
              role="checkbox"
              tabIndex="0"
              aria-checked="false"
            >
              <input
                checked={expirationByDate}
                onChange={handleExpirationTypeChange}
                type="checkbox"
                name="expiration[expires_at_strategy]"
                id="index_course_expiration_expires_at_strategy"
                data-role="expiration_expires_at_strategy"
              />
              <div className="checkbox" />
            </label>
            <p className="help-block">Pick the expiration date...</p>
          </div>
          { expirationByDate && (
            <div className="controls" id="expires_at_fields">
              <label htmlFor="expiration_expires_at">
                Expiration Date
              </label>
              <div className="controls">
                <DatePicker
                  selected={expirationDate}
                  onChange={handleExpirationDateChange}
                  name="expirationDate"
                  id="expiration_expires_at"
                  className="expires_at_datepicker picker__input"
                />
                <div className="picker" id="expiration_expires_at_root" aria-hidden="true" />
              </div>
            </div>
          )}
        </div>
        <div className="control-group featured expiration-strategy">
          <div className="controls">
            <label
              htmlFor="index_course_expiration_days_before_expire_strategy"
              className="label-switch"
              role="checkbox"
              tabIndex="0"
              aria-checked="true"
            >
              <input
                checked={expirationByDays}
                onChange={handleExpirationTypeChange}
                type="checkbox"
                name="expiration[days_before_expire_strategy]"
                id="index_course_expiration_days_before_expire_strategy"
                data-role="expiration_days_before_expire_strategy"
              />
              <div className="checkbox" />
            </label>
            <p className="help-block">...or select the number of days before expiration</p>
          </div>
          { expirationByDays && (
            <div className="controls" id="days_before_expire_fields">
              <label htmlFor="expiration_days_before_expire">
                Expires after
              </label>
              <div className="controls">
                <div className="expires-after-days-input-before">days</div>
                <input
                  value={expirationDays}
                  onChange={handleChange}
                  type="text"
                  name="expirationDays"
                  id="expiration_days_before_expire"
                  className="days_before_expire"
                />
              </div>
            </div>
          )}
        </div>
      </div>
    )}
  </fieldset>
);

AddCourseExpire.propTypes = {
  expirable: PropTypes.bool.isRequired,
  repurchasable: PropTypes.bool.isRequired,
  expirationByDate: PropTypes.bool.isRequired,
  expirationDate: PropTypes.instanceOf(Date),
  expirationByDays: PropTypes.bool.isRequired,
  expirationDays: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  handleExpirationTypeChange: PropTypes.func.isRequired,
  handleExpirationDateChange: PropTypes.func.isRequired
};

AddCourseExpire.defaultProps = {
  expirationDate: (() => {
    const tomorrow = new Date();
    tomorrow.setDate(tomorrow.getDate() + 1);
    return tomorrow;
  })(),
  expirationDays: undefined
};

export default AddCourseExpire;
