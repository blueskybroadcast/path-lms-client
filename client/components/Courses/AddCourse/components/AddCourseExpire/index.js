import AddCourseExpire from './component';

export default AddCourseExpire;
