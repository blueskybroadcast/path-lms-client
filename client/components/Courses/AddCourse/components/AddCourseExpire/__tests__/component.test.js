import React from 'react';
import { shallow } from 'enzyme';
import AddCourseExpire from '../component';

describe('AddCourseExpire', () => {
  const props = {
    handleChange: jest.fn(),
    handleExpirationTypeChange: jest.fn(),
    handleExpirationDateChange: jest.fn()
  };

  test('should render AddCourseExpire if it turned off correctly', () => {
    const wrapper = shallow(
      <AddCourseExpire
        {...props}
        expirable={false}
        repurchasable={false}
        expirationByDate
        expirationByDays={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render AddCourseExpire correctly if it turned on and expirationByDate enabled', () => {
    const wrapper = shallow(
      <AddCourseExpire
        {...props}
        expirable
        repurchasable
        expirationByDate
        expirationDate={new Date(Date.UTC(2019, 3, 12))}
        expirationByDays={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render AddCourseExpire correctly if it turned on and expirationByDays enabled', () => {
    const wrapper = shallow(
      <AddCourseExpire
        {...props}
        expirable
        repurchasable
        expirationByDate={false}
        expirationByDays
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
