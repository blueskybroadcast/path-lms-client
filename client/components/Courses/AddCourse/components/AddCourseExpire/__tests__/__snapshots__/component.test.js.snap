// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCourseExpire should render AddCourseExpire correctly if it turned on and expirationByDate enabled 1`] = `
<fieldset
  data-expirable="false"
  id="expirable_field"
>
  <legend>
    <span>
      Expiry Data
    </span>
  </legend>
  <div
    className="control-group featured"
  >
    <label
      htmlFor="index_course_expiration_expirable"
    >
      Expires
    </label>
    <div
      className="controls"
    >
      <label
        aria-checked="true"
        className="label-switch"
        htmlFor="index_course_expiration_expirable"
        role="checkbox"
        tabIndex="0"
      >
        <input
          checked={true}
          data-role="expiration_expirable"
          id="index_course_expiration_expirable"
          name="expirable"
          onChange={[MockFunction]}
          type="checkbox"
        />
        <div
          className="checkbox"
        />
      </label>
      <p
        className="help-block"
      >
        Access to this item expires
      </p>
    </div>
  </div>
  <div
    id="expiration_strategies_fields"
  >
    <div
      className="control-group featured"
    >
      <label
        htmlFor="index_course_expiration_repurchasable"
      >
        Repurchasable
      </label>
      <div
        className="controls"
      >
        <label
          aria-checked="true"
          className=" label-switch"
          htmlFor="index_course_expiration_repurchasable"
          role="checkbox"
          tabIndex="0"
        >
          <input
            checked={true}
            id="index_course_expiration_repurchasable"
            name="repurchasable"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <div
            className="checkbox"
          />
        </label>
        <p
          className="help-block"
        >
          This item can be repurchased after expiration
        </p>
      </div>
    </div>
    <div
      className="control-group featured expiration-strategy"
    >
      <div
        className="controls"
      >
        <label
          aria-checked="false"
          className="label-switch"
          htmlFor="index_course_expiration_expires_at_strategy"
          role="checkbox"
          tabIndex="0"
        >
          <input
            checked={true}
            data-role="expiration_expires_at_strategy"
            id="index_course_expiration_expires_at_strategy"
            name="expiration[expires_at_strategy]"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <div
            className="checkbox"
          />
        </label>
        <p
          className="help-block"
        >
          Pick the expiration date...
        </p>
      </div>
      <div
        className="controls"
        id="expires_at_fields"
      >
        <label
          htmlFor="expiration_expires_at"
        >
          Expiration Date
        </label>
        <div
          className="controls"
        >
          <DatePicker
            allowSameDay={false}
            className="expires_at_datepicker picker__input"
            dateFormat="MM/dd/yyyy"
            dateFormatCalendar="LLLL yyyy"
            disabled={false}
            disabledKeyboardNavigation={false}
            dropdownMode="scroll"
            id="expiration_expires_at"
            inlineFocusSelectedMonth={false}
            monthsShown={1}
            name="expirationDate"
            nextMonthButtonLabel="Next month"
            onBlur={[Function]}
            onChange={[MockFunction]}
            onClickOutside={[Function]}
            onFocus={[Function]}
            onInputClick={[Function]}
            onInputError={[Function]}
            onKeyDown={[Function]}
            onMonthChange={[Function]}
            onSelect={[Function]}
            onYearChange={[Function]}
            preventOpenOnFocus={false}
            previousMonthButtonLabel="Previous Month"
            readOnly={false}
            renderDayContents={[Function]}
            selected={2019-04-12T00:00:00.000Z}
            shouldCloseOnSelect={true}
            showTimeSelect={false}
            timeCaption="Time"
            timeIntervals={30}
            withPortal={false}
          />
          <div
            aria-hidden="true"
            className="picker"
            id="expiration_expires_at_root"
          />
        </div>
      </div>
    </div>
    <div
      className="control-group featured expiration-strategy"
    >
      <div
        className="controls"
      >
        <label
          aria-checked="true"
          className="label-switch"
          htmlFor="index_course_expiration_days_before_expire_strategy"
          role="checkbox"
          tabIndex="0"
        >
          <input
            checked={false}
            data-role="expiration_days_before_expire_strategy"
            id="index_course_expiration_days_before_expire_strategy"
            name="expiration[days_before_expire_strategy]"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <div
            className="checkbox"
          />
        </label>
        <p
          className="help-block"
        >
          ...or select the number of days before expiration
        </p>
      </div>
    </div>
  </div>
</fieldset>
`;

exports[`AddCourseExpire should render AddCourseExpire correctly if it turned on and expirationByDays enabled 1`] = `
<fieldset
  data-expirable="false"
  id="expirable_field"
>
  <legend>
    <span>
      Expiry Data
    </span>
  </legend>
  <div
    className="control-group featured"
  >
    <label
      htmlFor="index_course_expiration_expirable"
    >
      Expires
    </label>
    <div
      className="controls"
    >
      <label
        aria-checked="true"
        className="label-switch"
        htmlFor="index_course_expiration_expirable"
        role="checkbox"
        tabIndex="0"
      >
        <input
          checked={true}
          data-role="expiration_expirable"
          id="index_course_expiration_expirable"
          name="expirable"
          onChange={[MockFunction]}
          type="checkbox"
        />
        <div
          className="checkbox"
        />
      </label>
      <p
        className="help-block"
      >
        Access to this item expires
      </p>
    </div>
  </div>
  <div
    id="expiration_strategies_fields"
  >
    <div
      className="control-group featured"
    >
      <label
        htmlFor="index_course_expiration_repurchasable"
      >
        Repurchasable
      </label>
      <div
        className="controls"
      >
        <label
          aria-checked="true"
          className=" label-switch"
          htmlFor="index_course_expiration_repurchasable"
          role="checkbox"
          tabIndex="0"
        >
          <input
            checked={true}
            id="index_course_expiration_repurchasable"
            name="repurchasable"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <div
            className="checkbox"
          />
        </label>
        <p
          className="help-block"
        >
          This item can be repurchased after expiration
        </p>
      </div>
    </div>
    <div
      className="control-group featured expiration-strategy"
    >
      <div
        className="controls"
      >
        <label
          aria-checked="false"
          className="label-switch"
          htmlFor="index_course_expiration_expires_at_strategy"
          role="checkbox"
          tabIndex="0"
        >
          <input
            checked={false}
            data-role="expiration_expires_at_strategy"
            id="index_course_expiration_expires_at_strategy"
            name="expiration[expires_at_strategy]"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <div
            className="checkbox"
          />
        </label>
        <p
          className="help-block"
        >
          Pick the expiration date...
        </p>
      </div>
    </div>
    <div
      className="control-group featured expiration-strategy"
    >
      <div
        className="controls"
      >
        <label
          aria-checked="true"
          className="label-switch"
          htmlFor="index_course_expiration_days_before_expire_strategy"
          role="checkbox"
          tabIndex="0"
        >
          <input
            checked={true}
            data-role="expiration_days_before_expire_strategy"
            id="index_course_expiration_days_before_expire_strategy"
            name="expiration[days_before_expire_strategy]"
            onChange={[MockFunction]}
            type="checkbox"
          />
          <div
            className="checkbox"
          />
        </label>
        <p
          className="help-block"
        >
          ...or select the number of days before expiration
        </p>
      </div>
      <div
        className="controls"
        id="days_before_expire_fields"
      >
        <label
          htmlFor="expiration_days_before_expire"
        >
          Expires after
        </label>
        <div
          className="controls"
        >
          <div
            className="expires-after-days-input-before"
          >
            days
          </div>
          <input
            className="days_before_expire"
            id="expiration_days_before_expire"
            name="expirationDays"
            onChange={[MockFunction]}
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
</fieldset>
`;

exports[`AddCourseExpire should render AddCourseExpire if it turned off correctly 1`] = `
<fieldset
  data-expirable="false"
  id="expirable_field"
>
  <legend>
    <span>
      Expiry Data
    </span>
  </legend>
  <div
    className="control-group featured"
  >
    <label
      htmlFor="index_course_expiration_expirable"
    >
      Expires
    </label>
    <div
      className="controls"
    >
      <label
        aria-checked="true"
        className="label-switch"
        htmlFor="index_course_expiration_expirable"
        role="checkbox"
        tabIndex="0"
      >
        <input
          checked={false}
          data-role="expiration_expirable"
          id="index_course_expiration_expirable"
          name="expirable"
          onChange={[MockFunction]}
          type="checkbox"
        />
        <div
          className="checkbox"
        />
      </label>
      <p
        className="help-block"
      >
        Access to this item expires
      </p>
    </div>
  </div>
</fieldset>
`;
